autopilot/contractor/contractor.go

```diff
@@ -1057,3 +1007,26 @@ func performContractMaintenance(ctx *mCtx, alerter alerts.Alerter, s Database, c
 	// STEP 4: perform post maintenance tasks
 	return (nUpdated + nFormed) > 0, performPostMaintenanceTasks(ctx, s, alerter, cc, rb, logger)
 }
+
+// contractFunding is a helper that calculates the funding and collateral
+// that go into forming, refreshing or renewing a contract.
+func contractFunding(settings rhpv4.HostSettings, existingData uint64, minAllowance, minCollateral types.Currency, duration uint64) (allowance, collateral types.Currency) {
```

Comment:
Can't believe how this little helper makes the contract logic so much more readable and understandable.